projects/storefrontlib/src/cms-pages/product-details-page/product-details-url-matcher.ts

```diff
@@ -0,0 +1,40 @@
+import { inject, InjectionToken } from '@angular/core';
+import { Route } from '@angular/router';
+import {
+  DEFAULT_URL_MATCHER,
+  UrlMatcherFactory,
+  UrlMatcherService,
+} from '@spartacus/core';
+import { getSuffixUrlMatcher } from '../../cms-structure/routing/suffix-routes/suffix-url-matcher';
+
+export function getProductDetailsUrlMatcherFactory(
+  service: UrlMatcherService,
+  defaultMatcherFactory: UrlMatcherFactory
+): UrlMatcherFactory {
+  const factory = (route: Route) => {
+    const defaultMatcher = defaultMatcherFactory(route);
+    const suffixPDPMatcher = getSuffixUrlMatcher({
+      marker: 'p',
+      paramName: 'productCode',
+    });
+    return service.getCombined([defaultMatcher, suffixPDPMatcher]);
+  };
+  return factory;
+}
+
+/**
+ * Injection token with url matcher factory for PDP.
+ * The provided url matcher matches both:
+ * - the configured `paths` from routing config and
+ * - custom pattern  `** / p / :productCode`
```

Comment:
I left a comment regarding guarding a route, that's relevant here as well. Perhaps add it here as well, or link.